lockfile/src/spdx.rs

```diff
@@ -135,10 +138,18 @@ pub struct Spdx;
 
 impl Parse for Spdx {
     fn parse(&self, data: &str) -> anyhow::Result<Vec<Package>> {
-        let lock: SpdxInfo = serde_json::from_str(data).or_else(|_| serde_yaml::from_str(data))?;
+        let packages_info =
+            match serde_json::from_str::<SpdxInfo>(data).or_else(|_| serde_yaml::from_str(data)) {
+                Ok(lock) => lock.packages,
+                Err(_) => {
+                    let (_, b) =
+                        spdx::parse(data).finish().map_err(|e| anyhow!(convert_error(data, e)))?;
+                    b
+                },
+            };
```

Comment:
Totally optional, but I'd break this into a couple of separate statements to reduce nesting and improve readability a bit.

Reply:
I updated it. Not sure if there's a better way. I'd be happy to hear suggestions.

Reply:
I like the current solution. I was checking out for other combinators but seems like `Result` falls a bit short anyway.